apisix/plugins/traffic-split.lua

```diff
@@ -258,18 +258,21 @@ end
 local function new_rr_obj(weighted_upstreams)
     local server_list = {}
     for i, upstream_obj in ipairs(weighted_upstreams) do
-        if not upstream_obj.upstream then
+        if not upstream_obj.upstream and not upstream_obj.upstream_id then
             -- If the `upstream` object has only the `weight` value, it means
             -- that the `upstream` weight value on the default `route` has been reached.
-            -- Need to set an identifier to mark the empty upstream.
-            upstream_obj.upstream = "empty_upstream"
+            -- Mark empty upstream services in the plugin, but with weight.
+            upstream_obj.upstream = "plugin#upstream#is#empty"
+            server_list[upstream_obj.upstream] = upstream_obj.weight
```

Comment:
Could we unify the key type of `server_list`, we have two types of the key, i.e. string, table, and we have a special empty upstream mark.

Reply:
I don't think the key types of server_list need to be unified, because the keys of server_list can accept different types. If unified, it will bring additional data type conversion overhead.

Reply:
This is not good from the code readability perspective.